src/source-pcap-file.c

```diff
@@ -263,10 +263,9 @@ TmEcode ReceivePcapFileThreadInit(ThreadVars *tv, const void *initdata, void **d
             SCReturnInt(TM_ECODE_OK);
         }
 
+        pv->shared = &ptv->shared;
```

Comment:
I think this causes a double free on shared when status is not ok.

Reply:
@regit can you check this?

Reply:
When build and run with ASAN, it does not show a double free when inverting the clause (and log messages shows we are taking the error path).

When looking at the code, we have 2 cleaning functions call serially before returning:
- CleanupPcapFileFileVars: just set shared to NULL in pcap file strcuture
- CleanupPcapFileThreadVars: do some cleaning inside shared structure

So I think we are good.

Reply:
Works for me. Good to merge.